lib/pages/survey/survey_binding.dart

```diff
@@ -1,8 +1,13 @@
 import 'package:get/get.dart';
+import 'package:survey/api/graphql/graphql_client_provider.dart';
+import 'package:survey/repositories/survey_repository.dart';
+import 'package:survey/use_cases/get_survey_detail_use_case.dart';
 
 class SurveyBinding implements Bindings {
   @override
   void dependencies() {
-    // TODO: add DI binding
+    final graphQLClientProvider = Get.find<GraphQLClientProvider>();
+    final surveyRepository = SurveyRepositoryImpl(graphQLClientProvider.client);
```

Comment:
SurveyRepository is created once in HomeBinding. I think we can make SurveyRepository's scope as Singleton instead of creating a new instance in each screen. 🙏

Reply:
@manh-t 🤔 technically we already had the graphQLClient singleton 🤔  the key thing for the Repository is that in case we navigate on another flow that doesn't need the SurveyRepository, it occupies the memory for no purpose (even though it's small) 🤔  and for the consistency, should we make all the Repository becomes Singleton? 🤔  I think it's not necessary here unless we found a performance issue here 🤔

Reply:
FYI, in Android DI we are injecting them to the ViewModel scope level 🤔  I'm not sure if we could have scoping feature with GetX here 😔  but the idea is to reduce this configuration, we should be able to Get.find() without calling to put again, but NOT singleton 💇

Reply:
I found another way, but not a big fan of it tbh because under the hood it's still using Singleton persistence. 😔 
`lazyPut()` instead of `put()` will keep it re-useable.

Reply:
oh nope, `put()` will do, lazyPut() will only survive 1 time, so I better switch the UseCase to lazyPut() and keep the reuseable part in `put()` instead 😨

Reply:
> the key thing for the Repository is that in case we navigate on another flow (e.g: user profile) that doesn't need the SurveyRepository, it occupies the memory for no purpose (even though it's small) 

Yes, you're right. Technically I wanna bind the repository instance to the scope of specific flow, e.g non-authenticated or authenticated flow rather than making a repository as Singleton. But it seems not easily implement in Flutter, because the lifecycle of an instance depends on a widget tree 🤔

Reply:
> FYI, in Android DI we are injecting them to the ViewModel scope level 

In DM, we are injecting them to the activity scope 😂  And we have only one activity, so it's likely a Singleton 🤪 

> but the idea is to reduce this configuration, we should be able to Get.find() without calling to put again, but NOT singleton

I agree, let's go with this 🚀

Reply:
@sleepylee  I think we can make a class be Singleton or not through `permanent` parameter of `put()` method. 😅 
![image](https://user-images.githubusercontent.com/60863885/124584886-eee28400-de7e-11eb-8f69-b74a358ee92c.png)